src/transformers/modeling_bart.py

```diff
@@ -0,0 +1,1062 @@
+        encoder_outputs = _filter_out_falsey_values(encoder_outputs)  # type: tuple
+        return decoder_outputs + encoder_outputs
+
+
+class BartForMaskedLM(PretrainedBartModel):
+    r"""
+        **lm_labels**: (`optional`) ``torch.LongTensor`` of shape ``(batch_size, sequence_length)``:
+            Labels for computing the masked language modeling loss.
+            Indices should either be in ``[0, ..., config.vocab_size]`` or -100 (see ``input_ids`` docstring).
+            Tokens with indices set to ``-100`` are ignored (masked), the loss is only computed for the tokens with labels
+            in ``[0, ..., config.vocab_size]``.
+
+    Outputs: `Tuple` comprising various elements depending on the configuration (config) and inputs:
+        **loss**: (`optional`, returned when ``lm_labels`` is provided) ``torch.FloatTensor`` of shape ``(1,)``:
+            Masked language modeling loss.
+        **prediction_scores**: ``torch.FloatTensor`` of shape ``(batch_size, sequence_length, config.vocab_size)``
+            Prediction scores of the language modeling head (scores for each vocabulary token before SoftMax).
+        **hidden_states**: (`optional`, returned when ``config.output_hidden_states=True``)
+            list of ``torch.FloatTensor`` (one for the output of each layer + the output of the embeddings)
+            of shape ``(batch_size, sequence_length, hidden_size)``:
+            Hidden-states of the model at the output of each layer plus the initial embedding outputs.
+        **attentions**: (`optional`, returned when ``config.output_attentions=True``)
+            list of ``torch.FloatTensor`` (one for each layer) of shape ``(batch_size, num_heads, sequence_length, sequence_length)``:
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention heads.
+
+    Examples::
+
+        tokenizer = BartTokenizer.from_pretrained('bart-large')
+        model = BartForMaskedLM.from_pretrained('bart-large')
+        input_ids = torch.tensor(tokenizer.encode("Hello, my dog is cute")).unsqueeze(0)  # Batch size 1
+        outputs = model(input_ids=input_ids, lm_labels=input_ids)
+        loss, prediction_scores = outputs[:2]
+
+    """
+    base_model_prefix = "model"
+
+    def __init__(self, config: BartConfig):
+        super().__init__(config)
+        self.model = BartModel(config)
+        self.lm_head = _make_linear_from_emb(self.model.shared)
+
+    def forward(
+        self,
+        input_ids,
+        attention_mask=None,
+        encoder_hidden_states=None,
+        decoder_input_ids=None,
+        decoder_attention_mask=None,
+        cached_states=None,
+        lm_labels=None,
+        **unused
+    ):
+        outputs = self.model.forward(
+            input_ids,
+            attention_mask=attention_mask,
+            decoder_input_ids=decoder_input_ids,
+            encoder_hidden_states=encoder_hidden_states,
```

Comment:
@sshleifer - In BartModel the forward fn takes `encoder_outputs` as an input instead of `encoder_hidden_states`. 
Not a 100% sure, but shouldn't it be:
``
encoder_outputs=encoder_hidden_states
``
similar to how it is done for BartForSequenceClassificiation

Reply:
renamed everything encoder_outputs